### PR TITLE
Propose templating proof of concept

### PR DESCRIPTION
**Pull request by modpluz**
@lottspot, @crsmyz See the changes in this PR and let me know your thoughts.

**Notes:**
- Everything with regards to initial setup instructions of the base fork stays the same.
- After setup, you should then be able to navigate to `http://YOUR_HOST/candidates.html?slug=candidate-slug`.
- As at the time of this writing and as far as I know, only the candidate slug `lawrence-s-krasner` has data - @enactdev, correct me if I'm wrong on this.
- The implementations in this PR is also compatible with SEO friendly urls. e.g. `/candidates/lawrence-s-krasner`.

- Add the below to your nginx config to enable the feature mentioned above.
`rewrite ^/candidates/(.*)$ /candidate-detail.html?slug=$1 last;`

### CODE REVIEW

File: src/html/candidate-detail.html [hunk 1]
```diff
@@ -0,0 +1,30 @@
+<html>
+  <head>
+    <title>{{title}}</title>
```
Comment (lottspot) [inline src/html/candidate-detail.html]: Might have missed something, but I don't see where this actually gets rendered

File: src/js/candidate-detail.js [hunk 2]
```diff
@@ -0,0 +1,19 @@
+let urlParams = new URLSearchParams(window.location.search);
```
Comment (lottspot) [inline src/js/candidate-detail.js]: The `URLSearchParams` API [appears to have pretty narrow browser support](https://developer.mozilla.org/en-US/docs/Web/API/URLSearchParams#Browser_compatibility). I don't feel great about relying on it.

File: src/js/candidate-detail.js [hunk 3]
```diff
@@ -0,0 +1,19 @@
...
+let slug = urlParams.get('slug') || window.location.pathname.split('/')[2];
+// console.log(slug);
+// console.log(window.location.pathname);
```
Comment (lottspot) [inline src/js/candidate-detail.js]: can these just be deleted?

File: src/js/candidate-detail.js [hunk 4]
```diff
@@ -0,0 +1,19 @@
...
+
+let html = document.getElementById('mustache-template').innerHTML;
+let htmlContainer = 'pageContent';
```
Comment (lottspot) [inline src/js/candidate-detail.js]: Maybe we should add page specific sections to config.json where we could store stuff like this

File: src/js/candidate-detail.js [hunk 5]
```diff
@@ -0,0 +1,19 @@
...
+
+module.exports = (core) => {
+  core.getResource('candidates', { slug }).then((candidate) => {
+    console.log(candidate.data[0]);
+
+    candidate.data[0].name_full = `${candidate.data[0].name_first}  ${candidate.data[0].name_last}`;
+
+    let output = core.Mustache.render(html, { candidate: candidate.data[0] });
```
Comment (lottspot) [inline src/js/candidate-detail.js:13]: This is basically how we did the templating with the first frontend too. I'm ok with doing it this way 👍 what are your thoughts @crsmyz?

File: src/js/core.js [hunk 6]
```diff
@@ -1,13 +1,15 @@
 
-var fs = require('fs');
-var path = require('path');
-var request = require('request');
+const fs = require('fs');
```
Comment (lottspot) [inline src/js/core.js:2]: Thanks for ES6-ifying 😉

File: src/html/candidate-detail.html (hunk 1 above)
Comment (modpluz) [reply src/html/candidate-detail.html]: You're correct, this was left over from an earlier attempted approach and should be taken out.

File: src/js/candidate-detail.js (hunk 2 above)
Comment (modpluz) [reply src/js/candidate-detail.js]: Yes, I realized that. We can however, replace it with our own custom method - the implementations in this PR are more of a "proof of concept" if you will, from our discussion from yesterday and they are subjected to changes as we see fit.

File: src/js/candidate-detail.js (hunk 3 above)
Comment (modpluz) [reply src/js/candidate-detail.js]: Sure they can, another "left over".

File: src/js/candidate-detail.js (hunk 4 above)
Comment (modpluz) [reply src/js/candidate-detail.js]: Sure, I don't see why not.

File: src/js/candidate-detail.js (hunk 5 above)
Comment (modpluz) [reply src/js/candidate-detail.js:13]: Ah, I hadn't realized that.

File: src/js/core.js (hunk 6 above)
Comment (modpluz) [reply src/js/core.js:2]: You got it :)

File: src/js/candidate-detail.js (hunk 2 above)
Comment (lottspot) [reply src/js/candidate-detail.js]: Gotcha. As far as it being a PoC goes, I'm good with it

**Review by lottspot — APPROVED** [review]

File: src/js/candidate-detail.js (hunk 5 above)
Comment (medy-io) [reply src/js/candidate-detail.js:13]: I think its fine, especially if its just a poc

**Review by medy-io — APPROVED** [review]

**Pull request merged**